buildingBlocksOfOopPart1/src/Shape.ts

```diff
@@ -1,3 +1,49 @@
+import { Point } from "./Point";
+
 export abstract class Shape {
+  protected color: string;
+  protected filled: boolean;
+  protected points: Point[];
+
+  constructor(points: Point[]);
+  constructor(points: Point[], color: string, filled: boolean);
+  constructor(points: Point[], color?: string, filled?: boolean) {
+    if(points.length < 3) {
+      throw new Error(`[Points constructor]: received ${points.length} points, when minimum 3 points is required`)
+    }
+
+    if(color !== undefined && filled !== undefined) {
```

Comment:
this.color = color ?? 'green';
this.filled = filled ?? true;
looks prettier

Reply:
Fixed, thanks